services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -62,97 +62,180 @@ function hasTimestamps(fields) {
+              const manyToManyKeys = _.filter(constraints, (otherKey) =>
+                otherKey.columnName !== constraint.columnName
+                  && otherKey.columnType === 'FOREIGN KEY' && primaryKeys.includes(otherKey.columnName)) || [];
+
+              manyToManyKeys.forEach((manyToManyKey) => {
+                databaseSchema[refTableName].references.push(createReference(constraint, 'belongsToMany', manyToManyKey));
+              });
+              isManyToMany = manyToManyKeys !== [];
+            }
+          });
         }
-      });
+        if (!isManyToMany) {
+          const refPrimaryKeys = databaseSchema[refTableName].primaryKeys;
+          const refUniqueConstraint = databaseSchema[refTableName].constraints
+            .find(({ columnType }) => columnType === 'UNIQUE');
+          const refUniqueIndexes = refUniqueConstraint ? refUniqueConstraint.uniqueIndexes : null;
+          const refUnicity = checkUnicity(refPrimaryKeys, refUniqueIndexes, refColumnName);
 
-      const options = {
-        underscored: isUnderscored(fields),
-        timestamps: hasTimestamps(fields),
-        hasIdColumn: hasIdColumn(fields, primaryKeys),
-        hasPrimaryKeys: !_.isEmpty(primaryKeys),
-      };
-
-      return {
-        fields,
-        references,
-        primaryKeys,
-        options,
-      };
+          if (refUnicity.isPrimary || refUnicity.isUnique) {
+            table.references.push(createReference(constraint, 'belongsTo'));
+          }
+          databaseSchema[refTableName].references.push(
+            createReference(
+              constraint,
+              (isPrimary || isUnique) ? 'hasOne' : 'hasMany',
+            ),
+          );
+        }
+      }
     });
+  });
+}
+
+// NOTICE: Set the remaining fields
+async function createLianaSchema(tableSchema, tableName) {
+  const {
+    schema, constraints, primaryKeys, references,
+  } = tableSchema;
+  const fields = [];
+  await P.each(Object.keys(schema), async (columnName) => {
+    const columnInfo = schema[columnName];
+    const type = await columnTypeGetter.perform(columnInfo, columnName, tableName);
+    const foreignKey = _.find(constraints, { columnName, columnType: 'FOREIGN KEY' });
+    const isValidField = type && (!foreignKey
+      || !foreignKey.foreignTableName
+      || !foreignKey.columnName || columnInfo.primaryKey);
+
+    if (isValidField) {
+      // NOTICE: If the column is of integer type, named "id" and primary, Sequelize will
+      //         handle it automatically without necessary declaration.
+      if (columnName !== 'id' || type !== 'INTEGER' || !columnInfo.primaryKey) {
+        // NOTICE: Handle bit(1) to boolean conversion
+        let { defaultValue } = columnInfo;
+
+        if (defaultValue === "b'1'" || defaultValue === '((1))') {
+          defaultValue = true;
+        }
+        if (defaultValue === "b'0'" || defaultValue === '((0))') {
+          defaultValue = false;
+        }
```

Comment:
Small gift, well hidden in the refactoring.
No change description about that in the changelog :/